app/services/cart_services/check_out.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module CartServices
+  class CheckOut
+    def call(cart, order)
+      order.cart_items = cart.cart_items
+      if order.save
+        remove_items_from_cart order
+        OpenStruct.new(success?: true, payload: order)
+      else
+        OpenStruct.new(success?: false, payload: order)
+      end
+    end
+
+    private
+
+    def remove_items_from_cart(order)
+      order.cart_items.each do |item|
```

Comment:
poczytaj o transakcjach - to jest no-go. Zastanow sie jakie zagrozenie ta linijka moze powodowac

Reply:
Poprawione. Zrobię TIL o tym bo to poważna sprawa.